pyresample/test/test_dask_ewa.py

```diff
@@ -0,0 +1,389 @@
+                coords=output_coords,
+                dims=output_dims, attrs={'area': source_swath, 'test': 'test2',
+                                         'name': 'test2'})
+            new_data = resampler.resample(data, rows_per_scan=10)
+            new_data.compute()
+            # ll2cr will be called once more because of the computation
+            assert ll2cr.call_count == ll2cr_calls + num_chunks
+            # but we should already have taken the lonlats from the SwathDefinition
+            assert get_lonlats.call_count == lonlat_calls
+            _coord_and_crs_checks(new_data, target_area,
+                                  has_bands='bands' in input_dims)
+
+
+class TestDaskEWAResampler:
+    """Test Dask EWA resampler class."""
+
+    @pytest.mark.parametrize(
+        ('input_shape', 'input_dims', 'input_dtype', 'maximum_weight_mode'),
+        [
+            ((100, 50), ('y', 'x'), np.float32, False),
+            ((3, 100, 50), ('bands', 'y', 'x'), np.float32, False),
+            ((100, 50), ('y', 'x'), np.float32, True),
+            ((3, 100, 50), ('bands', 'y', 'x'), np.float32, True),
+            ((100, 50), ('y', 'x'), np.float64, False),
+            ((3, 100, 50), ('bands', 'y', 'x'), np.float64, False),
+            ((100, 50), ('y', 'x'), np.float64, True),
+            ((3, 100, 50), ('bands', 'y', 'x'), np.float64, True),
+        ]
+    )
+    def test_xarray_basic_ewa(self, input_shape, input_dims, input_dtype,
+                              maximum_weight_mode):
+        """Test EWA with basic xarray DataArrays."""
+        import xarray as xr
+        from pyresample.ewa import DaskEWAResampler, dask_ewa
+        output_shape = (200, 100)
+        if len(input_shape) == 3:
+            output_coords = {'bands': ['R', 'G', 'B']}
+            output_shape = (input_shape[0], output_shape[0], output_shape[1])
+            output_dims = ('bands', 'y', 'x')
+        else:
+            output_coords = {}
+            output_dims = ('y', 'x')
+        swath_data, source_swath, target_area = get_test_data(
+            input_shape=input_shape, output_shape=output_shape[-2:],
+            input_dims=input_dims, input_dtype=input_dtype,
+        )
+        # ignore column-wise chunks because DaskEWA should rechunk to use whole scans
+        num_chunks = len(source_swath.lons.chunks[0])
+
+        with mock.patch.object(dask_ewa, 'll2cr', wraps=dask_ewa.ll2cr) as ll2cr, \
+                mock.patch.object(source_swath, 'get_lonlats', wraps=source_swath.get_lonlats) as get_lonlats:
+            resampler = DaskEWAResampler(source_swath, target_area)
+            new_data = resampler.resample(swath_data, rows_per_scan=10,
+                                          maximum_weight_mode=maximum_weight_mode)
+            assert new_data.shape == output_shape
+            assert new_data.dtype == input_dtype
+            assert new_data.attrs['test'] == 'test'
+            assert new_data.attrs['area'] is target_area
```

Comment:
There duplicated code between this method and the next and the equivalent method in the previous class, could they be factorized? Also these methods are too long